### PR TITLE
Ensure that hex query predicates are normalized for planner cache

### PR DESCRIPTION
**Pull request by mattlord**
## Description
We were not normalizing hex values of either format:
- `0x01`
- `x'73757265'`

This had a major performance impact in workloads where hex query predicates are used in common queries as we could not leverage the query planner cache. This work normalizes them.

> Note: the only explicitly non-normalized values are now bit types of the format `b'1'`:
```
mysql> select b'011101110111010101110100' /* wut */;
+----------------------------------------------------------+
| b'011101110111010101110100'                              |
+----------------------------------------------------------+
| 0x777574                                                 |
+----------------------------------------------------------+
1 row in set (0.00 sec)
```

Should we try and handle those here as well???

## Related Issue(s)
https://github.com/vitessio/vitess/issues/9101

## Checklist
- [x] Should this PR be backported? Yes, at least to 12.0 IMO
- [x] Tests were added
- [x] Documentation is not required

### CODE REVIEW

**Comment by mattlord** [comment]
Making some progress: https://gist.github.com/mattlord/41575a480b667438e5e55ea3e740917b

File: go/test/endtoend/vtgate/queries/normalize/normalize_test.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	require.Nil(t, err)
+	found := false
+	for _, record := range results {
+		key := record["Key"].(string)
+		if key == normalizedInsertQuery {
+			found = true
+			break
+		}
+	}
+	assert.Equal(t, found, true, "correctly normalized record not found in planner cache")
```
Comment (systay) [inline go/test/endtoend/vtgate/queries/normalize/normalize_test.go:58]: nit: `assert.True` is nice

Comment (mattlord) [reply go/test/endtoend/vtgate/queries/normalize/normalize_test.go:58]: That is nicer, will switch to that. Thanks!

File: go/vt/sqlparser/ast_funcs.go [hunk 2]
```diff
@@ -480,6 +481,26 @@ func (node *Literal) HexDecode() ([]byte, error) {
 	return hex.DecodeString(node.Val)
 }
 
+// EncodeHexValToMySQLQueryFormat encodes the hexval back into the query format
+// for passing on to MySQL as a bind var
+func (node *Literal) EncodeHexValToMySQLQueryFormat() ([]byte, error) {
```
Comment (systay) [inline go/vt/sqlparser/ast_funcs.go:486]: does this really need to be a public method?

**Review by systay — APPROVED** [review]
Nice stuff

File: go/vt/sqlparser/ast_funcs.go (hunk 2 above)
Comment (mattlord) [reply go/vt/sqlparser/ast_funcs.go]: Good point. And, no, it doesn't. :) Done. Thanks!

File: java/client/pom.xml [hunk 3]
```diff
@@ -5,7 +5,7 @@
   <parent>
     <groupId>io.vitess</groupId>
     <artifactId>vitess-parent</artifactId>
-    <version>12.0.0-SNAPSHOT</version>
+    <version>13.0.0-SNAPSHOT</version>
```
Comment (frouioui) [inline java/client/pom.xml:8]: aren't these changes already applied by https://github.com/vitessio/vitess/pull/9066?

**Review by frouioui — APPROVED** [review]
this is very nice!

File: java/client/pom.xml (hunk 3 above)
Comment (mattlord) [reply java/client/pom.xml:8]: Yeah, not sure why it still shows them in the diff. I can rebase on main, but in the end this should not change either way when merged.

Comment (mattlord) [reply java/client/pom.xml]: rebased on main and they're gone from the diff.

**Pull request merged**